ElementX/Sources/Services/Users/UsersProviderProtocol.swift

```diff
@@ -0,0 +1,23 @@
+//
+// Copyright 2023 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+// sourcery: AutoMockable
+protocol UsersProviderProtocol {
+    func searchProfiles(with searchQuery: String) async -> [UserProfile]
+    func fetchSuggestions() -> [UserProfile]
+}
```

Comment:
```suggestion
protocol UsersProviderProtocol {
    func searchProfiles(with searchQuery: String) async -> Result<[UserProfile], UsersProviderError>
    func fetchSuggestions() async -> Result<[UserProfile], UsersProviderError>
}
```

Let's try to have a more generic signature to fulfill possible future requirements:
eg:
- Suggestions not sync
- Error handling